### PR TITLE
Add a bit of spec detail to the PROGRAM_BUILD_INFO_OPTIONS query.

### PR DESCRIPTION
**Pull request by aarongreig**
The new wording should make a bit more sense with the latest changes to the compiler API.

### CODE REVIEW

**Review by omarahmed1111 — APPROVED** [review]
LGTM

File: scripts/core/program.yml [hunk 1]
```diff
@@ -391,7 +391,7 @@ etors:
     - name: STATUS
       desc: "Program build status, return type $x_program_build_status_t"
     - name: OPTIONS
-      desc: "Program build options, return type char[]"
+      desc: "Options string specified by last build, compile or link operation performed on the program. Return type char[]"
```
Comment (kbenzie) [inline scripts/core/program.yml:394]: Should maybe be `char*` for the return type and mention it being null-terminated.

**Review by kbenzie — APPROVED** [review]
LGTM, one comment

File: scripts/core/program.yml (hunk 1 above)
Comment (aarongreig) [reply scripts/core/program.yml]: CL also does the [] thing but I don't think it makes a whole load of sense. I guess for char[] stuff it's like you're returning char[pPropSizeRet]? but that breaks down for stuff like int32_t[] so yeah changed all instances of that to use `*`

**Review by pbalcer — APPROVED** [review]

**Pull request merged**